File: backend/EnterpriseEmployeeSystem.sln/Employee.Api/Services/Products/ProductService.cs
using EnterpriseEmployeeSystem.Api.Models;
using EnterpriseEmployeeSystem.Api.Repositories.Products;

namespace EnterpriseEmployeeSystem.Api.Services.Products
{
    public class ProductService: IProductService
    {
        private readonly IProductRepository _productRepository;

        public ProductService(IProductRepository productRepository)
        {
            _productRepository = productRepository;
        }

        public async Task<Product?> GetActiveByItemCodeAsync(string itemCode)
        {
            return await _productRepository
                .GetActiveByItemCodeAsync(itemCode);
        }
    }
}

File: frontend/employee-ui/src/app/services/product.service.ts
import { inject, Service } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Product } from '../models/Product';

@Service()
export class ProductService {

  private http = inject(HttpClient);

  private apiUrl = '/api/products';

  getProduct(itemCode: string): Observable<Product> {
    return this.http.get<Product>(
      `${this.apiUrl}/${itemCode}`
    );
  }
}